src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionIterator.java

```diff
@@ -86,6 +88,8 @@ public AssemblyRegionIterator(final MultiIntervalShard<GATKRead> readShard,
         this.readCachingIterator = new ReadCachingIterator(readShard.iterator());
         this.readCache = new ArrayDeque<>();
         this.activityProfile = new BandPassActivityProfile(assemblyRegionArgs.maxProbPropagationDistance, assemblyRegionArgs.activeProbThreshold, BandPassActivityProfile.MAX_FILTER_SIZE, BandPassActivityProfile.DEFAULT_SIGMA, readHeader);
+// TODO: AH & BG handle changing contig
```

Comment:
@ahaessly can you elaborate what was missing WRT this TODO?

Reply:
Perhaps a simple test case that would cover the circumstance you were thinking of?

Reply:
This was something that james warned us about. I think the shards can be from different contigs so I think we needed to add code that checked for that case.